### DISCUSSION TITLE
'Download and run' progress animation starts (much) later on Pybricks Beta v2.0.0-beta.6 firmware

### DISCUSSION THREAD

**Discussion by BertLindeman** [pybricks/support#749] [General]
Not sure if this **is** a defect or feature or not, so started a discussion about this.

(Very nice that beta.pybricks.com now handles newer firmware again. Happy!)

The `download and run` button in ```v3.2.0b4 (Pybricks Beta v2.0.0-beta.6)```  takes longer to start the circle progress animation..
I often wonder if I **really** clicked the button.

As an example: it takes about 4 seconds after "click" for the progress circle comes up.
This is a program of just over 400 lines.
Assume this has to do with loading / including other sources.

**Remarkable**

1. Connected to Robot Inventor hub at beta6 firmware ```('primehub', '3.2.0b4', 'v3.2.0b4 on 2022-10-21')``` the button reacts very slow.

2. Connected to Spike Prime at CI firmware 2016  ```('primehub', '3.2.0b3', 'v3.2.0b3-55-gc126dd6b on 2022-08-26')``` this behaviour is "normal" and feels snappy.

**Comment by laurensvalk** [comment]
I agree. Would be great to start some form of progress indication right away. :+1: 

If it's hard to get meaningful numerical values, maybe we can do the same spinner we do for firmware updates? E.g. just spin for a while before it starts at 0%.

@BertLindeman: Would you mind sharing your script as a test-case. It was somewhat slow for me too but I haven't quite hit 4 seconds yet.

Also, would you mind posting this discussion (and also future things like these) as an issue? Thanks!

**Comment by BertLindeman** [comment]
New issue #750 created including the current python source.
